infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -298,10 +305,25 @@ private void handleException(Throwable ex, ChannelHandlerContext ctx, RequestMsg
      */
     @RequestHandler(type = PayloadCase.WRITE_LOG_REQUEST)
     private void handleWrite(RequestMsg req, ChannelHandlerContext ctx, IServerRouter router) {
+        final int uuidTailLength = 12;
+
         LogData logData = getLogData(req.getPayload().getWriteLogRequest().getLogData());
+        Map<UUID, Long> backPointerMap = logData.getBackpointerMap();
+
+        Map<String, Long> reducedBackPointerMap = new HashMap<>(backPointerMap.size());
+        backPointerMap.forEach((uuid, data) ->{
+            String uuidStr = uuid.toString();
+            String uuidTail = uuidStr.substring(uuidStr.length() - uuidTailLength);
```

Comment:
Today we only have a few thousands of streams at maximum. 32 digit UUID to 20 digit is good, but I suppose having the first 8 digits of the UUID should be just enough.

Reply:
I miscalculated the number. I feel like the last 8 digits will be hard to read. If you have a whole UUID string, the last 8 digits are not even a whole part separated by `-`. Imagine `f8c3de3d-1fea-4d7c-a8b0-29f63c4c3454` and last 8 digits will be `3c4c3454`. One the contrary `f8c3de3d` will be much easier to read.

Reply:
Sooooo, is 12 better?

Reply:
fixed, now it parses first 8 letters (it's more convenient)